src/terms/local_nonlinearity.jl

```diff
@@ -13,6 +13,7 @@ function ene_ops(term::TermLocalNonlinearity, basis::PlaneWaveBasis{T}, ψ, occu
                  ρ, kwargs...) where {T}
     fp(ρ) = ForwardDiff.derivative(term.f, ρ)
     E = sum(term.f.(ρ)) * basis.dvol
+    E = convert_dual(T, E)
```

Comment:
```suggestion
    E = sum(fρ -> convert_dual(T, fρ), term.f.(ρ)) * basis.dvol
```
has the advantage of doing less unneccessary computation in dual numbers and writing everything as one line.